cryptoprotocol/src/main/java/org/cryptimeleon/incentivesystem/cryptoprotocol/benchmark/BenchmarkConfig.java

```diff
@@ -0,0 +1,68 @@
+package org.cryptimeleon.incentivesystem.cryptoprotocol.benchmark;
+
+import lombok.AllArgsConstructor;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.IncentiveSystem;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.Setup;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.IncentivePublicParameters;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.provider.ProviderPublicKey;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.provider.ProviderSecretKey;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.user.UserPublicKey;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.user.UserSecretKey;
+
+/**
+ * A simple data class for a benchmark configuration with different constructors.
+ */
+@AllArgsConstructor
+public class BenchmarkConfig {
+    private static final Setup.BilinearGroupChoice DEFAULT_GROUP = Setup.BilinearGroupChoice.Herumi_MCL;
+    private static final int DEFAULT_SECURITY_PARAMETER = 128;
+
+    int iterations;
+    IncentiveSystem incentiveSystem;
+    IncentivePublicParameters pp;
+    ProviderPublicKey ppk;
+    ProviderSecretKey psk;
+    UserPublicKey upk;
+    UserSecretKey usk;
+
+    /**
+     * Create benchmark config with default configuration (see constant fields) with defined number of iterations.
+     * Generates fresh incentive system parameters and keys.
+     *
+     * @param iterations number of iterations for each protocol.
+     */
+    public BenchmarkConfig(int iterations) {
+        this.iterations = iterations;
+        manualSetup(DEFAULT_SECURITY_PARAMETER, DEFAULT_GROUP);
+    }
+
+    /**
+     * Create benchmark config with with defined number of iterations, security parameters and group.
+     * Generates fresh incentive system parameters and keys.
+     *
+     * @param iterations          number of iterations for each protocol.
```

Comment:
I was thinking whether it would be nice to set the #iterations individually for every protocol (would be useful if one of the protocols takes a lot of time while another one has a very fluctuating execution time, thus needing a lot of executions to get reliable numbers).
However, as far as I recall this does not seem to be the case when testing on PCs.

Reply:
Maybe a feature for the future, but I don't see many advantages as long as all protocols require about the same time